src/endpoints/Endpoint.js

```diff
@@ -86,20 +83,28 @@ class Endpoint {
     return `${this.stateName}Payload`
   }
 
-  fetchEndpointAction (urlParams = {}, options = {}) {
+  requestAction (urlParams = {}, options = {}) {
     return (dispatch, getState) => {
       if (getState()[this.name].isFetching) {
         return
       }
 
-      dispatch(this.requestAction())
-
-      let formattedURL = format(this.url, urlParams)
+      const formattedURL = format(this.url, urlParams)
       /*
        todo:  check if there are any paramters left in the url: formattedURL.match(/{(.*)?}/)
        currently this does not work as unused paramaters are just removed from the url
        */
 
+      const lastUrl = getState()[this.name].requestUrl
+
+      if (lastUrl && lastUrl === formattedURL) {
+        // fixme: Use "cached"
+        return
```

Comment:
If there is a cached version in the state use it instead of refetching. This needs a timeout.

Reply:
Timeout for what exactly ?

Reply:
So with this fix the data from the endpoint is only fetched once per page visit. Usually this shouldn't be a problem because the state is empty on each page visit but it can be cached by some extension of redux.

So maybe a lastFetchedDate state variable would be good.

Reply:
So those three lines need discussion. The first component which is displayed actually fetches the data and then the data from the state is used.

Reply:
Ah, so you mean a timespan for cache-expiration. (Timeout sounds like sth's going wrong :D ) Yes, a lastFetchedDate in the payload would be cool! I would suggest a span of 1-2 hours. Maybe we could even have a logic, that, when there's no internet connection and the cache has expired, we still use the cached version. (Even though we could probably also do this on service-worker lvl)
Furthermore (at least once the webapp runs in standalone) we can show a label saying when we last refreshed the content, when it's been loaded from cache.
Additionally it'd be cool, if the user can then force a refresh (meaning not invalidating, but only expiring the cache)

Reply:
I'm not really sure if it's worth it. I think it will be quite rare that that timeout gets triggered. I mean who has the app opened for more than 2 hours and expects the app to refresh after you have changed something in the cms?
You would just reload the page and then see the changes. So maybe the current implementation is enough

Reply:
I think there is just no use case.

Reply:
Well, I was talking about the case where we would have that Redux-Cache-Plugin-thingy in place. Then you can have content really cached over a long period and when you're offline the user could be unsure how up-to-date the content is.

Reply:
But then that discussion should be outside of this PR.